icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -129,28 +166,61 @@ int main(int argc, char* argv[]) {
         argc=-1;
     }
 
-    if(argc<0 || options[OPT_HELP_H].doesOccur || options[OPT_HELP_QUESTION_MARK].doesOccur) {
+    /* get the options values */
+    haveCopyright = options[OPT_COPYRIGHT].doesOccur;
+    destdir = options[OPT_DESTDIR].value;
+    VERBOSE = options[OPT_VERBOSE].doesOccur;
+    QUIET = options[OPT_QUIET].doesOccur;
+
+    // Load list of Unicode properties
+    std::vector<const char*> propNames;
+    for (int i=1; i<argc; i++) {
+        propNames.push_back(argv[i]);
+    }
+    if (options[OPT_ALL].doesOccur) {
+        for (int i=UCHAR_BINARY_START; i<UCHAR_INT_LIMIT; i++) {
+            if (i == UCHAR_BINARY_LIMIT) {
+                i = UCHAR_INT_START;
+            }
+            UProperty uprop = static_cast<UProperty>(i);
+            const char* propName = u_getPropertyName(uprop, U_SHORT_PROPERTY_NAME);
+            if (propName == NULL) {
+                propName = u_getPropertyName(uprop, U_LONG_PROPERTY_NAME);
+                if (propName != NULL && VERBOSE) {
+                    std::cerr << "Note: falling back to long name for: " << propName << std::endl;
+                }
+            }
+            if (propName != NULL) {
+                propNames.push_back(propName);
+            }
+        }
+    }
+
+    if(propNames.empty() || options[OPT_HELP_H].doesOccur || options[OPT_HELP_QUESTION_MARK].doesOccur || !options[OPT_MODE].doesOccur) {
         FILE *stdfile=argc<0 ? stderr : stdout;
         fprintf(stdfile,
-            "usage: %s [-options] properties...\n"
+            "usage: %s -m uprops [-options] [-A | properties...]\n"
             "\tdump Unicode property data to .toml files\n"
             "options:\n"
             "\t-h or -? or --help  this usage text\n"
             "\t-V or --version     show a version message\n"
-            "\t-c or --copyright   include a copyright notice\n"
+            "\t-m or --mode        mode: currently only 'uprops', but more may be added\n"
             "\t-t or --trie-type   set the trie type (small or fast, default small)\n"
             "\t-d or --destdir     destination directory, followed by the path\n"
+            "\t-A or --all         write out all properties known to icuexportdata\n"
+            "\t-I or --index       write an _index.toml summarizing all data exported\n"
```

Comment:
By the way, if you don't want to define a single-letter option, you can use something like `'\1'` in the UOPTION_DEF -- see other tools where I have done this. (Control code 1 is hard to type; and I don't advertise a single letter in the help text in such a case.)